aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -215,6 +218,34 @@ fn is_approved_gov_script(
     }
 }
 
+// TODO(George): move to config file when it is moved from types crate.
+pub fn aptos_prod_ty_builder(
```

Comment:
I think it is cleaner if we initialize type builder based on the gas feature version, features and gas params? Of course I can hardcode numbers, but the main issue is that MoveVM is not aware of gas params which carry dependency sizes, etc... should these be miscellaneous?

For now, I have the following:
1.  Feature disabled, use legacy,
2. Feature enabled, then:
    - If gas version >= 19 and gas params exist, use gas params,
    - Otherwise, use some default
  
Making it VM only concept requires an additional config, which has to communicate if I need to use legacy or limited type builder. If type builder is part of VM config, it should be exposed to apts VM so we can have prod configs with clear initializations?